### ISSUE TITLE
Feature: add flag to generate `Authentication` parameter into api interfaces

### ISSUE THREAD

**Issue by jamfor352** [kokuwaio/micronaut-openapi-codegen#218]
If I want to use micronaut-security's Authentication parameter, which is supported inside controller-level methods, it isn't clear how to do so with this generator.

For example, given I want to check the value of this using something like:

```
public reactor.core.publisher.Mono<io.micronaut.http.HttpResponse<MyObject>> mySpecialMethod(Authentication auth,
                                                          @PathVariable("someVariable") UUID someVariable,
                                                          @Body @Valid MyRequestBody request) {
        JwtUtils.mustBeProperlyDefined(auth);
        String username = auth.getAttributes().get("user_name").toString();
        <...Business logic here...>
}
```

None of the settings in the generator allow for adding the Authentication auth parameter (io.micronaut.security.authentication.Authentication) upon generation.

I've checked the documentation provided but this isn't clear at all.

**Comment by sschnabe** [comment]
Right now there is no flag for this.

To access the authentication you can use:

```java
@Inject
SecurityService security;

private Authentication getAuthentication() {
	return security.getAuthentication().orElseThrow(() -> new HttpStatusException(HttpStatus.UNAUTHORIZED, null));
}
```
Or use:

```java
private Authentication getAuthentication() {
	return ServerRequestContext.currentRequest()
			.flatMap(r -> r.getUserPrincipal(Authentication.class))
			.orElseThrow(() -> new HttpStatusException(HttpStatus.UNAUTHORIZED, null));
}
```

If this is not sufficient for you, how should the generator handle this authentication objects? My suggestion: add a flag and if flag is enabled and response code 401 exists add `Authentication` for controller interfaces. What do you think?

**Comment by jamfor352** [comment]
That's the workaround I'm using at the moment, it does work - it's just a bit verbose to have to add it into every controller, which adds up if there are multiple.

On another note - not sure if it's worth opening up a new issue - but how can I make sure from the API spec that a "number" input on an API Spec comes out as BigDecimal? I can see here we're explicitly mapping it to Double, but I don't see the reason for that https://github.com/kokuwaio/micronaut-openapi-codegen/blob/08793b1deda3826708c62f6a7b0225992a816efa/src/main/java/org/openapitools/codegen/languages/MicronautCodegen.java#L131

**Comment by sschnabe** [comment]
I'll add this flag, but not immediately because if other stuff todo before.

Regarding `BigDecimal` vs `Double`, you can override the default mapping using:

```xml
<typeMappings>
  <typeMapping>BigDecimal=java.math.BigDecimal</typeMapping>
</typeMappings>
```
The default with `Double` was chosen because for most users double precision is enough and `BigDecimal` handling is sometimes a pain. I won't change this now because a lot of users rely on `Double`.

**Comment by sschnabe** [comment]
Released with [3.3.0](https://github.com/kokuwaio/micronaut-openapi-codegen/releases/tag/3.3.0)
